subseas_flex_fcst/maproom.py

```diff
@@ -45,26 +48,75 @@
 
 APP.layout = layout.app_layout
 
-
-def read_cptdataset(y_transform=False):
-
-    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+#function to open all datasets and then combine them
+def combine_cptdataset(dataDir,filePattern,dof=False,y_transform=False):
+    filesNameList = glob.glob(f'{dataDir}/{filePattern}')
+    dataDic = {}
+    for idx, i in enumerate(filesNameList):
+        fileCompList = re.split('[_.]',filesNameList[idx])
+        fileCompList = [x for x in fileCompList if "/" not in x]
+
+        leadTimeWk = [x for x in fileCompList if x.startswith('wk')]
+
+        dateFormat = re.compile(".*-.*-.*")
+        startDateStr = list(filter(dateFormat.match,fileCompList))
+        startDate = startDateStr[0]
+        #startDate = datetime.strptime(startDateStr[0],
+        #'%b-%d-%Y').date()
+        if leadTimeWk[0] == 'wk1':
+            leadTimeDate = 'Week 1'#'Apr 2' #should it be made into dates or kept as wk number?
+        elif leadTimeWk[0] == 'wk2':
+            leadTimeDate = 'Week 2'#'Apr 9'
+        elif leadTimeWk[0] == 'wk3':
+            leadTimeDate = 'Week 3'#'Apr 16'
+        elif leadTimeWk[0] == 'wk4':
+            leadTimeDate = 'Week 4'#'Apr 23'
+        leadTimeDict = {'L':[leadTimeDate]}
+
+        ds = cptio.open_cptdataset(filesNameList[idx])
+        if dof == True:
+            dofVar = float(ds["tp_pred_err_var"].attrs["dof"])
+            ds = ds.assign(dof=dofVar)
+            #dofDict = {'dof':[dof]}
+            #ds =ds.assign_coords(dof=("dof",[dof]))
+            #ds = ds.expand_dims(dofDict)
+        if len(ds['T']) == 1:
```

Comment:
More needs to happen here: obs and hcst must be dealt with differently from mu and var. mu and var have one time dimension, whereas obs and hcst have two. mu and var time dimension is S, the date when the forecast is issued and that is organized by file and retrievable in the filename pattern. hcst and obs have two dimensions. The first one is the same S; the second one is a 78-point long time dimension which are the years of training. They are stored within each S-dependent file as a dimension of the xarray.

So we need to discriminate those 2 time dimensions. Right now they are mixed up. Good news is that the hcst/obs time dimension is already one provided when reading with open_cpt_dataset, so we need not create it of whave have you. Bad news that even though it's there, its values are crap. Good news is that it's crap the same way for hcst and obs, and hcst and obs only interact together before this time dimension is reduced, so it's fine to keep it as crap as it is.

So I would suggest (unless we want to clean up all the crappiness):

- for mu and var, after having read one file, remove the dummy 1-point T dimension that is there and then combine against S
- for obs and hcst: Don't touch their T dimension and combine also against S.

Reply:
And also, back to my more overarching comment, if we choose not to combine the data by S and L, but just use S/L to scrawl data files and make menus, then that problem would vanished on its own.

Reply:
For now, as you said, the problem solved itself. So this is an issue for a later date if we end up wanting to use the old function that combines all the data.